packages/utils/src/types/primitives.ts

```diff
@@ -196,18 +196,23 @@ export class Signature {
     if (signature.length !== SIGNATURE_LENGTH) {
       throw new Error('Incorrect size Uint8Array for signature')
     }
+    if (![0, 1].includes(recovery)) {
+      throw new Error('Recovery must be either 1 or 0, got ${recovery}')
+    }
   }
 
   static deserialize(arr: Uint8Array): Signature {
```

Comment:
Please document the use of https://eips.ethereum.org/EIPS/eip-2098 and how it applies to our code-based. I believe most reasoning is mentioned in the issue description.

Consider that within the ecosystem 65-bytes and 64-bytes are used, making it somewhat hard to grasp for someone looking at this and infer reasoning.

Reply:
We had a discussion with Robert about why we should keep the methods that use the 65-bytes encoding. The conclusion was that the `Signature` class does not need them anymore, so only 64-bytes signature encoding will be used from now on. There's a place where the legacy encoding is still present, but it's not related the context resolved by this PR.